### DISCUSSION TITLE
Update with lots of stuff - semi-stable

### DISCUSSION THREAD

**Discussion by bradhannah** [bradhannah/Ultima5Redux#333] [General]
This is a test release to confirm the execution flags are set correctly and save directories are created on all platforms.

<hr /><em>This discussion was created from the release <a href='https://github.com/bradhannah/Ultima5Redux/releases/tag/Aug-13-2022-PreAlpha'>Update with lots of stuff - semi-stable</a>.</em>

**Comment by bradhannah** [comment]
No hurry at all, I am working on lots of other features etc - but figured I would drop the latest Linux, Mac and Windows builds. Linux includes both OpenGL version and Vulkan. 

Tar'd up, so hopefully all the execution bits stick around on Mac and Linux this time too!

**Comment by bunnylin** [comment]
It runs! Performance seems tolerable, although some things take longer than I would reasonably expect. Most notably, when clicking to import a legacy save, the engine needs to think about it for a good 2 seconds before the list of importable files appears. Normally, I'd think it would be a split-second operation. Moving around in-game also has a slight delay, which becomes very apparent when trying to move quickly across the overworld and suddenly finding yourself in combat as the game graphics lagged behind and I never saw the enemy approach.

Looks like every in-game action gets a response of "GotActionSomething but no handler yet" even for actions that are clearly being handled, like moving and opening doors.

I'm able to light and extinguish torches, cast light spells. The scrollbar in the spellcasting menu still doesn't respond well to a mousewheel, mostly wiggling in place by pixel fractions instead of moving like a scrollbar should. There's nice music and sound effects. I see there are even SFX and Music volume sliders in the configuration menu, although they don't have an effect.

Combat has some nice UI now, looks quite workable. However, after attacking a bat I was stuck in targeting mode. It's unclear what's happening there, but the Turn counter keeps going up for every keypress. I think it's still trying to run the combat but the graphic tiles are no longer reflecting what's actually happening... By pressing sideways enough times, all party members fled combat and the game resumed on the overworld normally, except my party continued walking eastward as quickly as it could without any keypresses from me. They stopped once I did press the right cursor, suggesting a hanged keypress somehow occurred. The next combat with some orcs in a forest spawned on a completely black combat field with only the targeting rectangle visible. Happily the "Exit combat" button worked. I've also run into a troll and a squid.

**Reply by bradhannah** [reply]
Thanks for the feedback!!!
Re: import - yeah, there is a lot going on, been slowing optimizing it, but lots and lots of loops :) 

I am surprised about the slight delay. Think I may have to grab a copy of Parallels to test on Linux myself to see it. I experience zero lag in any of my tests so far. It has been a few weeks since I ran a profiler though. I find it's often one or two functions that create that kind of lag.

Re: GotActionSomething -> they are being handled, but aren't lol. Those handlers will include things such as visual cues on screen or sound effects.

Re: Scroll Wheel - I actually spent some time upping all the sensitivities and found them to be ideal - I suspect it's a Linux'ism I will again have to dig into. 

Re: Combat - there are some soft locks that crop up every once in a while. I wasn't aware of any still existing, but suppose I am not surprised. May need to a Debug dump hot key to help me know the state of the various important flags. For example you travel automatically in one direction after battle because a "AvatarIsMoving" flag gets stuck on - although I don't know why.

Lastly, re: odd match ups - would love to know more about the circumstance that put a troll and a squid together lol

**Reply by bunnylin** [reply]
The troll and squid were in separate fights. ^^; So that's working well, I think.
